src/QsCompiler/Core/Transformations/ExpressionTransformation.fs

```diff
@@ -461,6 +471,12 @@ and ExpressionTransformationBase internal (options: TransformationOptions, _inte
 
     // supplementary expression information
 
+    abstract OnLocalNameDeclaration : string -> string
+    default this.OnLocalNameDeclaration name = name
+
+    abstract OnLocalName : string -> string
+    default this.OnLocalName name = name
```

Comment:
It seems like if I want to visit callable parameters, or let/for statement bindings, I now need to override a handler in ExpressionTransformation, even if the declarations don't occur in an expression. Is that correct? I think that is confusing.

Reply:
I extracted these, and also all other shared pieces into the syntax tree transformation itself.